### ISSUE TITLE
Wish List Revamp

### ISSUE THREAD

**Issue by bhollis** [DestinyItemManager/DIM#10313]
This is an umbrella issue for the project of revamping wish lists. The end goal is:

1. Expand the capabilities of wish lists to better describe desired rolls, reduce redundancy, and allow for multiple wish list opinions on a single roll.
1. Have per-user personal wishlists that can be edited from within DIM's Armory UI (i.e. "save this roll to my wish list").
2. Store these wish lists in the DIM Sync system instead of requiring users to host text files in GitHub. Wish Lists can be shared with others directly from DIM Sync.

### Wish List Formats

Our existing text-based wish list format is repetitive and hard to author. We should replace it with something better:

* [x] Finalize https://github.com/DestinyItemManager/dim-api/pull/70
* [ ] Support Little Light's existing JSON wish list format
* [ ] https://github.com/DestinyItemManager/DIM/issues/10312
* [ ] https://github.com/DestinyItemManager/DIM/issues/7166
* [ ] Add conversion of old-style DIM wish lists into the new format. 
* [ ] Update LL's wishlist builder to support the extended features of our new format.

### Improvements to consuming wish lists

With the improved wish list schema, we can improve the experience of consuming wish lists.

* [ ] https://github.com/DestinyItemManager/DIM/issues/8893
* [ ] https://github.com/DestinyItemManager/DIM/issues/7128
* [ ] Display "wish list tags" and other metadata from the new wish list format on matched wish list rolls.
* [ ] Split the list of wish list rolls in the Armory view by various parameters (e.g. tag).
* [ ] Make the list of wish list rolls in the Armory view searchable / reorderable (e.g. by tag, date).
* [ ] In the Armory view's perk grid, show how many wish list rolls include each perk.
* [ ] Potentially introduce variations on the item tile's thumbs-up based on the roll's tag (e.g. "god roll")

### DIM API (DIM Sync)

For per-user personal wish lists to work, they need to be saved into DIM Sync so that users don't have to manually host them.

* [ ] Design a SQL schema for personal wish lists. This should support efficient partial updates (i.e. doesn't upload the whole wish list every time) and be efficiently delta-syncable (so we don't have to re-download the whole wish list every time).
* [ ] Implement DIM Sync APIs for bulk uploading wish lists, updating individual items/rolls, and downloading the wish list.
* [ ] Build dim.gg share URLs for sharing wish lists with others. This should be tied to an opt-in "public" setting on the wish list.
* [ ] Allow for users to have multiple wish lists they "own" - for example users may want to have a public, shareable list and a private one for their own use. Or they might want to publish a different list per season.
* [ ] Potentially support versioning of wish lists, and/or a "draft" and "publish" system so users can iterate on wish lists before publishing them.
* [ ] Upload existing wish lists (voltron, etc.) to DIM Sync as public wish lists.

### Wish List Editor

Users need the ability to add and remove individual wish listed rolls to their wish list, as well as to see all the rolls in their wish list in one place.

* [ ] The primary interface for users to add rolls to their personal wish list (or any other selected wish list they own) should be the Armory page:  https://github.com/DestinyItemManager/DIM/issues/8629
* [ ] Create a new wish list view page that can show all the rolls from a wish list, organized by weapon. This should be a searchable list of all items that are mentioned in the wish list, and clicking on the item should bring up the Armory page, with its wish list section pre-filtered to show only rolls from that wish list. Of course, that Armory page would also allow editing the wish list if you own it. This is also where you can set wish lists to public/private, etc.

**Comment by Revadike** [comment]
>  Wishlists seem terribly inconvenient to maintain. They should be better integrated, as opposed to requiring external tools. Allow us to select the preferred perks directly on the weapon, when inspecting them. It would also be nice to define multiple god roll combo's or variations.

**Comment by Revadike** [comment]
Excited for this revamp!

**Comment by TheCodeGnome** [comment]
Saw this flagged as `Help Wanted: Developer`. Do you have any criteria for someone to help out on the project, @bhollis, or is it an open call?

**Comment by bhollis** [comment]
It's open for whoever wants to work on it. Tasks are listed out above if you want to start making progress on it!

**Comment by TheCodeGnome** [comment]
Very cool. I'll have a look and see if I can contribute anything other than more bugs.

**Comment by Revadike** [comment]
> Very cool. I'll have a look and see if I can contribute anything other than more bugs.

Did you get anywhere?

**Comment by bhollis** [comment]
When there are updates they'll be noted right here, or in the changelog

**Comment by TheCodeGnome** [comment]
Unfortunately, no. Been busy with work.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by aselke** [comment]
If/when this gets worked on, it would be handy to be able to add a piece of Armor 3.0 gear to the wishlist.  It could be as simple as specifying an exotic name or armor set piece, with a preferred archetype and optionally a minimum tier value.

In the case of exotic class items, it would be great to also be able to specify the exotic perk combination.

**Comment by bhollis** [comment]
Wish lists already support exotic class items, but will not be extended to general armor. Those are easily found with search.

Also, this issue is not for general comments or requests for wish lists, please file those as their own issues.

**Comment by SolaneHub** [comment]
Hello, I think I'll probably work to some if not all issues here. I'm free on the weekend so that's my free time.
I worked on the Little Light JSON format, but I'd like to know which features have priorities over others, and if some features have to be added first than others. Hopefully D2 doesn't die.

**Comment by SolaneHub** [comment]
https://github.com/SolaneHub/DIM/tree/multi-wishlist-support

I'm leaving this here. I was tired of the wishlist displaying only the first god roll available completely ignoring combinations and not combining rolls into 1 display in the armory. Now that ANY weapon and future weapons weapons will drop with 3 perks in the most relevant column (3+4, sometimes 5) the god roll indicator can't wait anymore. We can't wait to a have a full wishlist revamp for this. I have dozens of copies of the same weapon and going through them with the current system is awful. I'm using this in my own local copy of DIM. If u want take what I did, if not peace.

**Comment by Revadike** [comment]
yes, I want my godrolls to include multiple perks. I am no longer grinding for single perk combinations.

**Comment by Revadike** [comment]
@bhollis with the game ending, do you think we'll see this issue ever get resolved still or is there no chance?
